tests/unit/test-cmd/test.build.js

```diff
@@ -37,6 +44,94 @@ describe('build', () => {
     );
   });
 
+  it('gives the correct name to an extension that is not localized', () => {
+    return withTempDir(
+      (tmpDir) =>
+        build({
+          sourceDir: fixturePath('minimal-web-ext'),
+          artifactsDir: tmpDir.path(),
+        })
+        .then((buildResult) => {
+          assert.match(buildResult.extensionPath,
+                       /minimal_extension-1\.0\.zip$/);
+          return buildResult.extensionPath;
+        })
+    );
+  });
+
+  it('gives the correct name to a localized extension', () => {
+    return withTempDir(
+      (tmpDir) =>
+        build({
+          sourceDir: fixturePath('minimal-localizable-web-ext'),
+          artifactsDir: tmpDir.path(),
+        })
+        .then((buildResult) => {
+          assert.match(buildResult.extensionPath,
+                       /name_of_the_extension-1\.0\.zip$/);
+          return buildResult.extensionPath;
+        })
+    );
+  });
+
+  it('checks locale file for malformed json', () => {
+    return withTempDir(
+      (tmpDir) => {
+        const messageFileName = path.join(tmpDir.path(), 'messages.json');
+        fs.writeFileSync(messageFileName,
+                '{"simulated:" "json syntax error"');
+        return getDefaultLocalizedName({
+          messageFile: messageFileName,
+          manifestData: manifestWithoutApps,
+        })
+        .then(makeSureItFails())
+        .catch((error) => {
+          assert.equal(
+            error instanceof WebExtError, true);
+          assert.match(error.message, /The JSON file is malformed/);
```

Comment:
I would like to see some of the caught and re-thrown error here to test that the code includes the original error, like:
````
 /The JSON file is malformed: the original error.../
````